File: rectangle-js/rectangle.js
// your code, here
class Rectangle {
  constructor(width, height) {
    this.width = width;
    if (height === undefined) {
      this.height = width;
    }
    else {
      this.height = height;
    }
  }

  area() {
    return this.height * this.width;
  }
}

File: rectangle_constructor/rectangle.rb
class Rectangle
  attr_reader :width, :height

  def initialize width, height = nil
    @width = width
    if height
      @height = height
    else
      @height = nil
    end
  end

  def area
    return @width ** 2 unless height
    return @width * @height
  end
end
